man/melt.data.table.Rd

```diff
@@ -87,52 +87,58 @@ An unkeyed \code{data.table} containing the molten data.
 set.seed(45)
 require(data.table)
 DT <- data.table(
-      i_1 = c(1:5, NA),
-      i_2 = c(NA,6,7,8,9,10),
-      f_1 = factor(sample(c(letters[1:3], NA), 6, TRUE)),
-      f_2 = factor(c("z", "a", "x", "c", "x", "x"), ordered=TRUE),
-      c_1 = sample(c(letters[1:3], NA), 6, TRUE),
-      d_1 = as.Date(c(1:3,NA,4:5), origin="2013-09-01"),
-      d_2 = as.Date(6:1, origin="2012-01-01"))
+  i_1 = c(1:5, NA),
+  n_1 = c(NA, 6, 7, 8, 9, 10),
+  f_1 = factor(sample(c(letters[1:3], NA), 6L, TRUE)),
+  f_2 = factor(c("z", "a", "x", "c", "x", "x"), ordered=TRUE),
+  c_1 = sample(c(letters[1:3], NA), 6L, TRUE),
+  c_2 = sample(c(LETTERS[1:2], NA), 6L, TRUE),
+  d_1 = as.Date(c(1:3,NA,4:5), origin="2013-09-01"),
+  d_2 = as.Date(6:1, origin="2012-01-01")
+)
 # add a couple of list cols
-DT[, l_1 := DT[, list(c=list(rep(i_1, sample(5,1)))), by = i_1]$c]
-DT[, l_2 := DT[, list(c=list(rep(c_1, sample(5,1)))), by = i_1]$c]
+DT[, l_1 := DT[, list(c=list(rep(i_1, sample(5, 1L)))), by = i_1]$c]
+DT[, l_2 := DT[, list(c=list(rep(c_1, sample(5, 1L)))), by = i_1]$c]
 
-# id, measure as character/integer/numeric vectors
-melt(DT, id=1:2, measure="f_1")
-melt(DT, id=c("i_1", "i_2"), measure=3) # same as above
-melt(DT, id=1:2, measure=3L, value.factor=TRUE) # same, but 'value' is factor
-melt(DT, id=1:2, measure=3:4, value.factor=TRUE) # 'value' is *ordered* factor
+# id.vars, measure.vars as character/integer/numeric vectors
+melt(DT, id.vars=1:2, measure.vars="f_1")
+melt(DT, id.vars=c("i_1", "n_1"), measure.vars=3) # same as above
+melt(DT, id.vars=1:2, measure.vars=3L, value.factor=TRUE) # same, but 'value' is factor
+melt(DT, id.vars=1:2, measure.vars=3:4, value.factor=TRUE) # 'value' is *ordered* factor
 
 # preserves attribute when types are identical, ex: Date
-melt(DT, id=3:4, measure=c("d_1", "d_2"))
-melt(DT, id=3:4, measure=c("i_1", "d_1")) # attribute not preserved
+melt(DT, id.vars=3:4, measure.vars=c("d_1", "d_2"))
+melt(DT, id.vars=3:4, measure.vars=c("n_1", "d_1")) # attribute not preserved
 
 # on list
-melt(DT, id=1, measure=c("l_1", "l_2")) # value is a list
-melt(DT, id=1, measure=c("c_1", "l_1")) # c1 coerced to list
+melt(DT, id.vars=1, measure.vars=c("l_1", "l_2")) # value is a list
+suppressWarnings(
+  melt(DT, id.vars=1, measure.vars=c("c_1", "l_1")) # c1 coerced to list, with warning
+)
 
 # on character
-melt(DT, id=1, measure=c("c_1", "f_1")) # value is char
-melt(DT, id=1, measure=c("c_1", "i_2")) # i2 coerced to char
+melt(DT, id.vars=1, measure.vars=c("c_1", "f_1")) # value is char
+suppressWarnings(
+  melt(DT, id.vars=1, measure.vars=c("c_1", "n_1")) # n_1 coerced to char, with warning
+)
 
 # on na.rm=TRUE. NAs are removed efficiently, from within C
-melt(DT, id=1, measure=c("c_1", "i_2"), na.rm=TRUE) # remove NA
+melt(DT, id.vars=1, measure.vars=c("c_1", "c_2"), na.rm=TRUE) # remove NA
```

Comment:
change to `c_2` (also added this column) to focus on `na.rm`, otherwise there's a type promotion warning as well